common/beerocks/bcl/source/network/get_iface_stats.cpp

```diff
@@ -0,0 +1,243 @@
+    io.iov_len           = req.hdr.nlmsg_len;
+    rtnl_msg.msg_iov     = &io;
+    rtnl_msg.msg_iovlen  = 1;
+    rtnl_msg.msg_name    = &socket;
+    rtnl_msg.msg_namelen = sizeof(socket);
+
+    if (sendmsg(fd, static_cast<struct msghdr *>(&rtnl_msg), 0) < 0) {
+        LOG(ERROR) << "Unable to send message through Netlink socket: " << strerror(errno);
+    } else {
+        bool msg_done = false; /* flag to end loop parsing */
+
+        /**
+         * Parse reply until message is done
+         */
+        while (!msg_done) {
+            struct msghdr rtnl_reply {
+            }; /* generic msghdr structure for use with recvmsg */
+
+            /* a large buffer to receive lots of link information */
+            char reply[IFLIST_REPLY_BUFFER];
+
+            io.iov_base            = reply;
+            io.iov_len             = IFLIST_REPLY_BUFFER;
+            rtnl_reply.msg_iov     = &io;
+            rtnl_reply.msg_iovlen  = 1;
+            rtnl_reply.msg_name    = &socket;
+            rtnl_reply.msg_namelen = sizeof(socket);
+
+            /**
+             * Read as much data as fits in the receive buffer
+             */
+            int length = recvmsg(fd, &rtnl_reply, 0);
+            for (const struct nlmsghdr *msg_ptr = reinterpret_cast<const struct nlmsghdr *>(reply);
+                 NLMSG_OK(msg_ptr, length); msg_ptr = NLMSG_NEXT(msg_ptr, length)) {
+                switch (msg_ptr->nlmsg_type) {
+                case NLMSG_DONE:
+                    /**
+                     * This is the special meaning NLMSG_DONE message we asked for by using NLM_F_DUMP flag
+                     */
+                    msg_done = true;
+                    break;
+                case RTM_NEWLINK:
+                    /**
+                     * This is a RTM_NEWLINK message, which contains lots of information about a link
+                     */
+                    if (get_iface_stats(msg_ptr, iface_name, iface_stats)) {
+                        msg_done = true;
+                        result   = true;
+                    }
+                    break;
+                }
+            }
+        }
+    }
+
+    return result;
+}
+
+/**
+ * @brief Gets interface statistics for the given network interface.
+ *
+ * Gets interface stats for given network by means of a Netlink socket using NETLINK_ROUTE
+ * protocol.
+ *
+ * @param[in] iface_name Name of the network interface.
+ * @param[in, out] iface_stats Interface statistics structure with values read.
+ *
+ * @return True on success and false otherwise.
+ */
+static bool get_iface_stats(const std::string &iface_name, sInterfaceStats &iface_stats)
+{
+    bool result = false;
+
+    /**
+     * Create Netlink socket for kernel/user-space communication.
+     * No need to call bind() as packets are sent only between the kernel and the originating
+     * process (no multicasting).
+     */
+    int fd = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+    if (fd < 0) {
```

Comment:
from the [CODINGSTYLE.md:](https://github.com/prplfoundation/prplMesh/blob/master/CODINGSTYLE.md)
`Bad - Single point of exit which adds an unecessary indentation level to the else clause, and makes the function longer:`

Reply:
Done